playground/frontend/playground_components/lib/src/enums/complexity.dart

```diff
@@ -25,18 +25,16 @@ enum Complexity {
   medium,
   @JsonValue('ADVANCED')
   advanced,
-  ;
+}
 
-  static Complexity? fromString(String? complexity) {
-    switch (complexity) {
-      case 'basic':
-        return Complexity.basic;
-      case 'medium':
-        return Complexity.medium;
-      case 'advanced':
-        return Complexity.advanced;
-    }
+extension ComplexityFromString on String {
```

Comment:
I like the static method better.
- With `str.toCompexity()` it is harder to locate the implementation than with `Complexity.fromString(str)`.
- `Complexity.fromString` suggests the return type more clearly.
- A String is wide-used, and it should not be "aware" of specific conversions to tons of classes we may come up with.